### PR TITLE
Map targets to multiple interfaces and Migrations

### PR DESCRIPTION
**Pull request by ryanmelt**
Map targets to multiple interfaces
Migrations. 
Sync Control Code support. 
Remove as_config from models.

### CODE REVIEW

File: openc3/lib/openc3/models/interface_model.rb [hunk 1]
```diff
@@ -281,37 +295,59 @@ def undeploy
...
+      microservice.target_names.delete(target_name) unless @target_names.include?(target_name)
       microservice.update
     end
 
-    def map_target(target_name)
+    def map_target(target_name, cmd_only: false, tlm_only: false, unmap_old: true)
+      if cmd_only and tlm_only
+        cmd_only = false
+        tlm_only = false
+      end
```
Comment (jmthomas) [inline openc3/lib/openc3/models/interface_model.rb:331]: How could you ever have cmd_only AND tlm_only? That doesn't logically make sense.

File: openc3/lib/openc3/interfaces/interface.rb [hunk 2]
```diff
@@ -133,6 +139,8 @@ def initialize
       @name = self.class.to_s.split("::")[-1] # Remove namespacing if present
       @state = 'DISCONNECTED'
       @target_names = []
```
Comment (jmthomas) [inline openc3/lib/openc3/interfaces/interface.rb:141]: We're keeping @target_names for now or is this deprecated?

**Review by jmthomas — APPROVED** [review]
Looks good! Should this be part of the Demo some way?

File: openc3/lib/openc3/models/interface_model.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/models/interface_model.rb:331]: Nothing keeping a user from setting both cmd_only and tlm_only so I handled that case.

File: openc3/lib/openc3/interfaces/interface.rb (hunk 2 above)
Comment (ryanmelt) [reply openc3/lib/openc3/interfaces/interface.rb:141]: Keeping it for backwards compatibility.  Its a superset of cmd_target_names and tlm_target_names.   Generally only the newer lists should be used, but if you just want to know all targets supported, target_names still saves you having to merge the other two lists and uniq.

**Pull request merged**